FLEDGE.md

```diff
@@ -180,9 +194,15 @@ const myAuctionConfig = {
                         '*': 150,
                         ...},
   'perBuyerGroupLimits': {'https://www.example-dsp.com': 2,
-                        'https://www.another-buyer.com': 1000,
-                        '*': 15,
-                        ...},
+                          'https://www.another-buyer.com': 1000,
+                          '*': 15,
+                          ...},
+  'perBuyerPrioritySignals': {'https://www.example-dsp.com': {'signal1': 2.5,
```

Comment:
I was wondering, perhaps `perBuyerPrioritySignals` could be a field within `perBuyerSignals['https://www.example-dsp.com']`? That way, the protocol between the seller and the buyer wouldn't have to be updated to handle the priority signals. We'd also lose the natural support for the default (wildcard) priority signals (but currently, unless I'm missing something, I don't see a good use for them).

Reply:
I'm a bit concerned about overloading fields (field A is both data sent directly to buyer scripts, and data used to run these filters).  Seems like there's a lot of room for unexpected behaviors there.  If someone, for instance, is using a field we suddenly decide has another meaning, we'd be breaking consumers.

If we want to include with in perBuyerSignals, I'd be more comfortable structuring perBuyerSignals as:
{
  arbitraryJsonData: <...>
  prioritySignals: {...}
}

And dropping everything that's not in one of those two values, to minimize the chance of confusion.  That would, of course, be a breaking change, but so would trying to extract priority signals from perBuyerSignals without the format change.

Reply:
There was some related discussion on this here:
https://github.com/WICG/turtledove/pull/266#discussion_r836612673

The direction discussed was leaving `perBuyerSignals` alone, letting it contain arbitrary buyer json, but introducing a new `perBuyerConfigurations` field that would contain the various buyer-specific fields that FLEDGE needs to handle, e.g. `prioritySignal`, `experimentGroupId`, `perBuyerGroupLimits`, and potentially `interestGroupBuyers`.

Reply:
That's a very different proposal - at least I assume jonasz's suggestion was so that the signals would be passed to worklets, as opposed to be a suggestion on how to better organize fields.

Reply:
Ah, my read of 
> That way, the protocol between the seller and the buyer wouldn't have to be updated to handle the priority signals

was 

"sellers can avoid redeploying code every time FLEDGE adds a new per-buyer field if the per-buyer fields are consolidated"

I don't see that a `perBuyerConfigurations` map clashes with making the signals available to worklets -- actually I'd think it desirable to make `perBuyerConfigurations` available to the worklet so that things like `experimentGroupId` and `prioritySignals` don't have to be redundantly passed through `perBuyerSignals` if they're needed by the worklet.

Reply:
> That's a very different proposal - at least I assume jonasz's suggestion was so that the signals would be passed to worklets, as opposed to be a suggestion on how to better organize fields.

My proposal was actually purely technical, as @jrmooring summarized.

I agree that if there are other use cases where the buyer needs to pass some configuration to the auction, something like `perBuyerConfigurations` would be nice.

Reply:
I suspect we're going to switch to something along the lines of `perBuyerConfigurations` when we end the Fledge field trial and feel we can more safely break backwards compatibility.  We could also switch to passing in the entire configuration to bidder worklets at the same time.